ddl/ddl_api.go

```diff
@@ -471,7 +471,10 @@ func checkDefaultValue(ctx sessionctx.Context, c *table.Column, hasDefaultValue
 		if types.ErrTruncated.Equal(err) {
 			return types.ErrInvalidDefault.GenByArgs(c.Name)
 		}
-		return errors.Trace(err)
+		if err != nil {
```

Comment:
Why not combine it with the upper `if`?

Reply:
@lamxTyler  It's a bit complicate here I found.  Illegal default value doesn't necessary result in `ErrInvalidDefault`, actually. For example:

In MySQL:
```sql
mysql> create table t(a decimal(3) default 1.25567);
Query OK, 0 rows affected, 1 warning (0.03 sec)

mysql> show warnings;
+-------+------+----------------------------------------+
| Level | Code | Message                                |
+-------+------+----------------------------------------+
| Note  | 1265 | Data truncated for column 'a' at row 1 |
+-------+------+----------------------------------------+
1 row in set (0.00 sec)
```

In TiDB:
```sql
tidb> create table t(a decimal(3) default 1.25567);
ERROR 1067 (42000): Invalid default value for 'a'

```

But if default value check if set in `ALTER TABLE` situation(ddl/ddl_db_test.go: line 1139), The 
`ErrInvalidDefault` error is raised. 

another example:

Im MySQL:
```sql
mysql> create table t ( a date default '2012-01-01 01:01:01');
Query OK, 0 rows affected, 1 warning (0.02 sec)

mysql> show warnings;
+-------+------+---------------------------------------------------------------------+
| Level | Code | Message                                                             |
+-------+------+---------------------------------------------------------------------+
| Note  | 1292 | Incorrect date value: '2012-01-01 01:01:01' for column 'a' at row 1 |
+-------+------+---------------------------------------------------------------------+
1 row in set (0.00 sec)
```

In TiDB:
```sql
create table t ( a date default '2012-01-01 01:01:01');
Query OK, 0 rows affected (0.01 sec)
```

I think in case  #6303  , we can fix the error message by throwing out `ErrInvalidDefault ` in all means, and Fix the corner cases or situations in extra one or more prs ..